common/list.c

```diff
@@ -62,6 +62,13 @@ void list_cat(list_t *list, list_t *source) {
 	}
 }
 
+void list_empty(list_t *list) {
```

Comment:
I'd be much less subtle and just set list->length to 0, list_add does reallocs but list_del never bothers with reducing capacity either so meh.